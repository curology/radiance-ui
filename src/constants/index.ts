export { default as ANIMATION } from './animation';
export { default as BREAKPOINTS } from './breakpoints';
export { default as GRID } from './grid';
export { default as LAYOUT } from './layout';
export { default as MEDIA_QUERIES } from './mediaQueries';
export * from './portals';
export { default as PROGRESS_BAR_STATUS } from './progressBarStatus';
export { default as SPACER } from './spacer';
export * from './themes';
export { default as Z_SCALE } from './zScale';
