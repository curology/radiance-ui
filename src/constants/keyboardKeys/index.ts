export const keyboardKeys = {
  enter: 'Enter',
} as const;
