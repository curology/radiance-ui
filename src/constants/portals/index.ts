export const REACT_PORTAL_SECTION_ID = 'reactPortalSection';
