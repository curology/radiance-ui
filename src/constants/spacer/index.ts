const SPACER = {
  /**
   * 2px
   */
  x2small: '0.125rem',
  /**
   * 4px
   */
  xsmall: '0.25rem',
  /**
   * 8px
   */
  small: '0.5rem',
  /**
   * 16px
   */
  base: '1rem',
  /**
   * 16px
   */
  medium: '1rem',
  /**
   * 24px
   */
  large: '1.5rem',
  /**
   * 32px
   */
  xlarge: '2rem',
  /**
   * 40px
   */
  x2large: '2.5rem',
  /**
   * 48px
   */
  x3large: '3rem',
  /**
   * 56px
   */
  x4large: '3.5rem',
  /**
   * 64px
   */
  x5large: '4rem',
} as const;

export default SPACER;
