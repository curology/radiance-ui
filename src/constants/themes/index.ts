export { primaryTheme } from './primaryTheme';
export { secondaryTheme } from './secondaryTheme';
export { tertiaryTheme } from './tertiaryTheme';
export * from './types';
