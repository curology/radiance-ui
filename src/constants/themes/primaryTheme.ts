import PRIMARY_COLORS from '../colors/primary';
import PRIMARY_BORDER_RADIUS from '../borderRadius/primary';
import PRIMARY_BOX_SHADOWS from '../boxShadows/primary';
import PRIMARY_FONTS from '../fonts/primary';
import { PRIMARY_TYPOGRAPHY } from '../typography/primary';
import { LogoPrimary } from '../../icons/logos/svgs';

export const primaryTheme = {
  __type: 'primary',
  BORDER_RADIUS: PRIMARY_BORDER_RADIUS,
  BOX_SHADOWS: PRIMARY_BOX_SHADOWS,
  COLORS: PRIMARY_COLORS,
  FONTS: PRIMARY_FONTS,
  TYPOGRAPHY: PRIMARY_TYPOGRAPHY,
  LOGO: LogoPrimary({}),
} as const;
