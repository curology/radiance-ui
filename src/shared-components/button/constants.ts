import type { ThemeType } from '../../constants';

// TODO: potentially break out these pairings to the color constants file
export const textColorsAssociatedWithColors = (theme: ThemeType) =>
  new Map([
    [
      theme.COLORS.primary,
      {
        tint1: theme.COLORS.primaryTint1,
        tint2: theme.COLORS.primaryTint2,
      },
    ],
    [
      theme.COLORS.secondary,
      {
        tint1: theme.COLORS.secondaryTint1,
        tint2: theme.COLORS.secondaryTint2,
      },
    ],
    ['other', undefined],
  ]);
