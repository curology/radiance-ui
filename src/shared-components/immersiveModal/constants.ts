export const MODAL_MOBILE_SCROLLING_ID = 'modal-mobile-scrolling-id';
export const MODAL_DESKTOP_SCROLLING_ID = 'modal-desktop-scrolling-id';
export const CROSS_ICON_CONTAINER_ID = 'cross-icon-container';
