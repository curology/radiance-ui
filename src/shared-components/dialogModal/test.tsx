import React from 'react';
import { primaryTheme } from 'src/constants/themes';
import { render } from 'src/tests/testingLibraryHelpers';

import { DialogModal } from './index';

const modalTitle = 'Dialog Modal Title';
const bodyString = 'Dialog Modal Children Content';
const modalBody = <DialogModal.Paragraph>{bodyString}</DialogModal.Paragraph>;

describe('<DialogModal />', () => {
  /**
   * TODO: Fix Emotion 11 CI snapshot serializer order issue
   */
  // eslint-disable-next-line jest/no-disabled-tests
  describe.skip('UI snapshots', () => {
    it('renders dialog modal with custom color', async () => {
      const { container, findByText } = render(
        <DialogModal backgroundColor={primaryTheme.COLORS.background}>
          <div>{modalBody}</div>
        </DialogModal>,
        { withPortalContainer: true },
      );

      await findByText(bodyString);

      expect(container).toMatchSnapshot();
    });
  });

  it('render children content correctly', async () => {
    const { getAllByText, findByText } = render(
      <DialogModal title={modalTitle}>{modalBody}</DialogModal>,
      { withPortalContainer: true },
    );

    const body = await findByText(bodyString);

    expect(getAllByText(modalTitle).length > 0).toBe(true);
    expect(body).toBeInTheDocument();
  });
});
