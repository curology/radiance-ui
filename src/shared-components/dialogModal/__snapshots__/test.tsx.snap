// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogModal /> UI snapshots renders dialog modal with custom color 1`] = `
.emotion-0 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 999999;
  background-color: rgba(45, 45, 48, 0.7);
  -webkit-transition: opacity 250ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: opacity 250ms cubic-bezier(0.075, 0.82, 0.165, 1);
}

.emotion-0.entering,
.emotion-0.exiting,
.emotion-0.exited {
  opacity: 0;
}

.emotion-0.entered {
  opacity: 1;
}

@media (min-width: 768px) {
  .emotion-0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-flex-flow: row nowrap;
    -webkit-flex-flow: row nowrap;
    -ms-flex-flow: row nowrap;
    flex-flow: row nowrap;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    -webkit-justify-content: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

.emotion-2 {
  width: 100%;
  margin: 0 auto;
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  border-top-left-radius: 32px;
  border-top-right-radius: 32px;
  box-shadow: 0px -8px 24px rgba(52, 51, 82, 0.05);
  background: #F8F8FA;
  padding: 3.5rem 1.5rem 2rem;
  overflow-y: auto;
  max-height: 100%;
  -webkit-transition: -webkit-transform 250ms cubic-bezier(0.075, 0.82, 0.165, 1);
  transition: transform 250ms cubic-bezier(0.075, 0.82, 0.165, 1);
}

.emotion-2.entered {
  -webkit-transform: translateY(0%);
  -moz-transform: translateY(0%);
  -ms-transform: translateY(0%);
  transform: translateY(0%);
}

.emotion-2.entering,
.emotion-2.exiting,
.emotion-2.exited {
  -webkit-transform: translateY(100%);
  -moz-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
}

@media (min-width: 576px) {
  .emotion-2 {
    max-height: 700px;
  }
}

@media (min-width: 768px) {
  .emotion-2 {
    position: relative;
    width: 456px;
    border-radius: 8px;
    padding: 3.5rem;
  }

  .emotion-2.entering,
  .emotion-2.exiting,
  .emotion-2.exited {
    -webkit-transform: translateY(40%);
    -moz-transform: translateY(40%);
    -ms-transform: translateY(40%);
    transform: translateY(40%);
  }
}

.emotion-4 {
  position: absolute;
  top: 8px;
  right: 12px;
  z-index: 2000;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  background: #F8F8FA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: auto;
  cursor: pointer;
}

@media (min-width: 768px) {
  .emotion-4 {
    top: 16px;
    right: 16px;
  }
}

.emotion-4:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-7 {
  margin-bottom: 1.5rem;
}

.emotion-7:last-of-type {
  margin-bottom: 2rem;
}

<div
  id="reactPortalSection"
>
  <div
    class="entering emotion-0 emotion-1"
  >
    <span
      data-focus-scope-start="true"
      hidden=""
    />
    <div
      class="entering emotion-2 emotion-3"
      role="alertdialog"
      tabindex="-1"
    >
      <div
        aria-label="Close modal"
        class="emotion-4 emotion-5"
        role="button"
        tabindex="0"
      >
        <svg
          class="emotion-6"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m16.01.93-.89-.88-7.11 7.1L.91.05.02.93l7.11 7.11-7.11 7.11.89.88 7.1-7.11 7.11 7.11.89-.88L8.9 8.04 16.01.93Z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div>
        <p
          class="emotion-7 emotion-8"
        >
          Dialog Modal Children Content
        </p>
      </div>
    </div>
    <span
      data-focus-scope-end="true"
      hidden=""
    />
  </div>
</div>
`;
